src/Shared/RequestMethodHelper.cs

```diff
@@ -0,0 +1,77 @@
+// <copyright file="RequestMethodHelper.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if NET8_0_OR_GREATER
+using System.Collections.Frozen;
+#endif
+
+namespace OpenTelemetry.Internal;
+
+internal static class RequestMethodHelper
+{
+#if NET8_0_OR_GREATER
+    internal static readonly FrozenDictionary<string, string> KnownMethods;
+#else
+    internal static readonly Dictionary<string, string> KnownMethods;
+#endif
+
+    static RequestMethodHelper()
+    {
+#if NET8_0_OR_GREATER
+        KnownMethods = FrozenDictionary.ToFrozenDictionary(
+            new[]
+            {
+                KeyValuePair.Create("GET", "GET"),
+                KeyValuePair.Create("PUT", "PUT"),
+                KeyValuePair.Create("POST", "POST"),
+                KeyValuePair.Create("DELETE", "DELETE"),
+                KeyValuePair.Create("HEAD", "HEAD"),
+                KeyValuePair.Create("OPTIONS", "OPTIONS"),
+                KeyValuePair.Create("TRACE", "TRACE"),
+                KeyValuePair.Create("PATCH", "PATCH"),
+                KeyValuePair.Create("CONNECT", "CONNECT"),
+            },
+            StringComparer.OrdinalIgnoreCase);
+#else
+        KnownMethods = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "GET", "GET" },
+            { "PUT", "PUT" },
+            { "POST", "POST" },
+            { "DELETE", "DELETE" },
+            { "HEAD", "HEAD" },
+            { "OPTIONS", "OPTIONS" },
+            { "TRACE", "TRACE" },
+            { "PATCH", "PATCH" },
+            { "CONNECT", "CONNECT" },
+        };
+#endif
+    }
+
+    public static bool TryResolveHttpMethod(string method, out string resolvedMethod)
+    {
+        if (KnownMethods.TryGetValue(method, out resolvedMethod))
```

Comment:
@vishweshbankwar Would you do a bit of benchmarking here? I think a `switch (method)` might actually be the fastest thing to do here (even faster than frozen dictionary).

Check out: https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-8/#frozen-collections

If you scroll down from that anchor to the "IsMostPopular" example, the compiler generates really smart logic for a string switch. I'm guessing it will be faster because it is customized for the known values and done inline, plus it also doesn't need the processing at startup FrozenDictionary needs.

Reply:
@CodeBlanch We need a case insensitive search here. I think we would end up doing either `ToUpper` or `ToLower` if we want to use a switch case. That could again add to the cost.

Reply:
@CodeBlanch We need case-insensitive look up. https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-8/#frozen-collections:~:text=such%20that%20if%20you%20know%20all%20of%20your%20string%20keys%20at%20compile%2Dtime%2C%20and%20you%20just%20need%20an%20ordinal%2C%20case%2Dsensitive%20lookup%2C%20you%20might%20be%20best%20off%20simply%20writing%20a%20switch%20statement%20or%20expression

Reply:
Makes sense however I still _feel_ like we could solve it in a way that would be faster. Given there are only a handful of values and we know them all upfront. Perf challenge for anyone willing to take it on 😄